vfio/src/vfio_pci.rs

```diff
@@ -311,6 +311,7 @@ pub struct VfioPciDevice {
     mmio_regions: Vec<MmioRegion>,
     interrupt: Interrupt,
     interrupt_routes: Vec<InterruptRoute>,
+    viommu_attached: bool,
```

Comment:
One nit: `iommu_attached`, not `viommu`.

Although I know this is all quite PCI bound, I think it would make more sense to add that boolean to `VfioDevice` instead. And then check for `iommu_attached` directly from `VfioDevice`'s `unset_dma_map()` and `setup_dma_map()` so that it all becomes transparent for the `VfioPciDevice` implementation.

Reply:
Yes good point, done!